Plan/src/main/java/com/djrapitops/plan/system/database/databases/sql/H2DB.java

```diff
@@ -0,0 +1,276 @@
+import com.djrapitops.plugin.task.AbsRunnable;
+import com.djrapitops.plugin.task.PluginTask;
+import com.djrapitops.plugin.task.RunnableFactory;
+import dagger.Lazy;
+import org.h2.jdbcx.JdbcDataSource;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.File;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Objects;
+
+/**
+ * Implementation of the H2 database
+ *
+ * @author Fuzzlemann
+ * @since 4.5.1
+ */
+public class H2DB extends SQLDB {
+
+    private final File databaseFile;
+    private final String dbName;
+    private Connection connection;
+    private PluginTask connectionPingTask;
+
+    private H2DB(
+            File databaseFile,
+            Locale locale,
+            PlanConfig config,
+            Lazy<ServerInfo> serverInfo,
+            NetworkContainer.Factory networkContainerFactory,
+            RunnableFactory runnableFactory,
+            PluginLogger logger,
+            Timings timings,
+            ErrorHandler errorHandler
+    ) {
+        super(() -> serverInfo.get().getServerUUID(), locale, config, networkContainerFactory, runnableFactory, logger, timings, errorHandler);
+        dbName = databaseFile.getName();
+        this.databaseFile = databaseFile;
+    }
+
+    @Override
+    public void setupDataSource() throws DBInitException {
+        try {
+            Class.forName("org.h2.Driver");
+        } catch (ClassNotFoundException e) {
+            errorHandler.log(L.CRITICAL, this.getClass(), e);
+        }
+
+        try {
+            connection = getNewConnection(databaseFile);
+        } catch (SQLException e) {
+            throw new DBInitException(e);
+        }
+
+        execute("SET REFERENTIAL_INTEGRITY FALSE");
+        startConnectionPingTask();
+    }
+
+    public Connection getNewConnection(File dbFile) throws SQLException {
+        String dbFilePath = dbFile.getAbsolutePath();
+
+        Connection newConnection = getConnectionFor(dbFilePath);
+        logger.debug("H2 " + dbName + ": Opened a new Connection");
+        newConnection.setAutoCommit(false);
+        return newConnection;
+    }
+
+    private Connection getConnectionFor(String dbFilePath) throws SQLException {
+        String username = config.getString(Settings.DB_USER);
+        String password = config.getString(Settings.DB_PASS);
+
+        JdbcDataSource jdbcDataSource = new JdbcDataSource();
+        jdbcDataSource.setURL("jdbc:h2:file:" + dbFilePath + ";mode=MySQL");
+        jdbcDataSource.setUser(username);
+        jdbcDataSource.setPassword(password);
+
+        return jdbcDataSource.getConnection();
+    }
+
+    private void startConnectionPingTask() {
+        stopConnectionPingTask();
+        try {
+            // Maintains Connection.
+            connectionPingTask = runnableFactory.create("DBConnectionPingTask " + getName(), new AbsRunnable() {
```

Comment:
Could this `AbsRunnable` be split into a new class, since it is now in use in two different places? :)

Reply:
Refactored in ab61f16